lib/slayer/command.rb

```diff
@@ -6,37 +6,51 @@ class Command
     class << self
       def call(*args, &block)
         # Run the Command and capture the result
-        result = new.tap { |s|
-          s.run(*args, &block)
-        }.result
+        command = self.new
+        result  = command.tap { |s| s.run(*args) }.result
+
+        # Run block
+        yield(result, command) if block_given?
+
+        # Throw an exception if we don't return a result
+        raise CommandNotImplemented unless result.is_a? Result
```

Comment:
I feel like CommandNotImplemented isn't the right Error to raise here. Thoughts?